core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1651,13 +1652,22 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
    *
    * <p>({@code TO_CHAR} is not supported in MySQL, but it is supported in
    * MariaDB, a variant of MySQL covered by {@link SqlLibrary#MYSQL}.) */
-  @LibraryOperator(libraries = {MYSQL, ORACLE, POSTGRESQL})
+  @LibraryOperator(libraries = {MYSQL, ORACLE})
   public static final SqlFunction TO_CHAR =
       SqlBasicFunction.create("TO_CHAR",
-          ReturnTypes.VARCHAR,
+          ReturnTypes.VARCHAR_NULLABLE,
           OperandTypes.TIMESTAMP_STRING,
           SqlFunctionCategory.TIMEDATE);
 
+  /** The "TO_CHAR(timestamp, format)" function;
+   * converts {@code timestamp} to string according to the given {@code format}. */
+  @LibraryOperator(libraries = {POSTGRESQL})
+  public static final SqlFunction TO_CHAR_PG =
+      new SqlBasicFunction("TO_CHAR", SqlKind.OTHER_FUNCTION,
```

Comment:
Can this be used directly?
```
SqlBasicFunction.create("TO_CHAR",
          ReturnTypes.VARCHAR_NULLABLE,
          OperandTypes.TIMESTAMP_STRING,
          SqlFunctionCategory.TIMEDATE);
```

Reply:
No. The implementations of `hashCode` and `equals` are from `SqlOperator` and only looks at the name and kind.

There is also a `TO_CHAR` for MySQL/Oracle and this new one for PostgreSQL. Since they have the same name and kind they would conflict in a HashMap. Using a new class here gets around the collision. Another solution is to add `equals` and `hashCode` implementations to `SqlBasicFunction`.

Reply:
Understood. 

A small suggestion, I think the `SqlBasicFunction` constructor is designed to be as private as possible. Is it possible to add a create method to create the corresponding anonymous class, for example:

```
  public static SqlBasicFunction overloadCreate(String name,
      SqlReturnTypeInference returnTypeInference,
      SqlOperandTypeChecker operandTypeChecker, SqlFunctionCategory category) {
    return new SqlBasicFunction(name, SqlKind.OTHER_FUNCTION,
        SqlSyntax.FUNCTION, true, returnTypeInference, null,
        OperandHandlers.DEFAULT, operandTypeChecker, 0,
        category, call -> SqlMonotonicity.NOT_MONOTONIC, false){};
  }
```